scripts/conda2requirements.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This file parses a conda file and outputs a requirements.txt file with the same libraries
+
+import argparse
+import yaml
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Converter from conda.yml to requirements.txt",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("-i", "--input", help="Input filename", default="conda.yaml")
+    parser.add_argument(
+        "-o", "--output", help="Output filename", default="requirements.txt"
+    )
+    args = parser.parse_args()
+
```

Comment:
if you want to remove the step for generating the yaml file and pull the dicts directly then you can do something like this:
```
    all_deps = CONDA_BASE
    all_deps.update(CONDA_PYSPARK)
    all_deps.update(CONDA_GPU)
    all_deps.update(PIP_BASE)
    all_deps.update(PIP_PYSPARK)
    all_deps.update(PIP_GPU)
    all_deps.update(PIP_DARWIN)
    all_deps.update(PIP_LINUX)
    all_deps.update(PIP_WIN32)

    with open(args.output, "w") as f:
        f.write("\n".join([v for k, v in all_deps.items()]))
```

Reply:
one downside of both of these approaches though, is we lose the cpu based packages, we should add some logic in either approach to generate entries for duplicate keys e.g. (tensorflow -> tensorflow, tensorflow-gpu)

Reply:
I changed the functionality